src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,183 @@
+            if (keywords.length == 1) throw new MissingParameterException("Search");
+            handleSearch(keywords[1]);
+            break;
+          case "done":
+            if (keywords.length == 1) throw new MissingParameterException("Done");
+            handleDone(keywords[1]);
+            break;
+          case "delete":
+            if (keywords.length == 1) throw new MissingParameterException("Delete");
+            handleDelete(keywords[1]);
+            break;
+          case "todo":
+            if (keywords.length == 1) throw new MissingParameterException("Todo");
+            handleTodo(keywords[1]);
+            break;
+          case "event":
+            if (keywords.length == 1) throw new MissingParameterException("Event");
+            handleEvent(keywords[1]);
+            break;
+          case "deadline":
+            if (keywords.length == 1) throw new MissingParameterException("Deadline");
+            handleDeadline(keywords[1]);
+            break;
+          default:
+            throw new UnknownCommandException(keywords[0]);
+        }
+
+        storage.saveBaby(taskList.getTaskList());
+
+      } catch (MissingParameterException
+          | MissingTimeException
+          | UnknownCommandException
+          | TimeFormatException
+          | InvalidIndexException e) {
+        ui.showException(e);
+      } catch (Exception e) {
+        ui.showUnknownException(e);
+      }
+    }
+    ui.bye();
+  }
+
+  /**
+   * Handles the searching for key in the task list
+   * @param key String that we want to look for in the task.
+   */
+  public void handleSearch(String key) {
+    ArrayList<Task> tasksWithKey = new ArrayList<>();
+    for (Task t: taskList.getTaskList()) {
+      if (t.toString().contains(key)) {
+        tasksWithKey.add(t);
+      }
+    }
+    ui.showList(tasksWithKey, "Here are the tasks that contains '" + key + "' in your list:");
+  }
+
+  /**
+   * Marks task as done and call Ui to display task.
+   *
+   * @param indexStr A string representing the index of the task from list we want to mark as done.
+   */
+  public void handleDone(String indexStr) {
+    try {
+      int index = Integer.parseInt(indexStr) - 1;
+      Task task = taskList.getTaskList().get(index);
+      task.done();
+      ui.showDone(task);
+    } catch (Exception e) {
+      throw new InvalidIndexException(indexStr);
+    }
+  }
+
+  /**
+   * Deletes tasks from task list and call Ui to display deleted task.
+   *
+   * @param indexStr A string representing the index of the task from list we want to delete.
+   */
+  public void handleDelete(String indexStr) {
+    try {
+      int index = Integer.parseInt(indexStr) - 1;
+      Task task = taskList.getTaskList().get(index);
+      taskList.getTaskList().remove(task);
+      ui.showDelete(task, taskList.getTaskList());
+    } catch (Exception e) {
+      throw new InvalidIndexException(indexStr);
+    }
+  }
+
+  /**
+   * Handle the event command
```

Comment:
A bit of nitpicking, but Handle should be Handles, as grammatical errors can have a large impact when documenting professionally in future!